aw_client/client.py

```diff
@@ -255,65 +255,44 @@ def __init__(self, client, dispatch_interval=0):
+        self._current = None  # type: Optional[QueuedRequest]
+
+    def _get_next(self) -> Optional[QueuedRequest]:
+        # self._current will always hold the next not-yet-sent event,
+        # until self._task_done() is called.
+        if not self._current:
+            try:
+                self._current = self._persistqueue.get(block=False)
+            except persistqueue.exceptions.Empty:
+                return None
+        return self._current
+
+    def _task_done(self) -> None:
+        self._current = None
+        self._persistqueue.task_done()
+
+    def _create_buckets(self) -> None:
         # Check if bucket exists
         buckets = self.client.get_buckets()
         for bucket in self._registered_buckets:
             if bucket.id not in buckets:
                 self.client.create_bucket(bucket.id, bucket.type)
 
-    def _queue_to_file(self, endpoint: str, data: dict):
-        entry = QueuedRequest(endpoint=endpoint, data=data)
-        with open(self.queue_file, "a+") as queue_fp:
-            queue_fp.write(json.dumps(entry) + "\n")
-
-    def _load_queue(self):
-        # If crash when lost connection, queue failed requests
-        failed_requests = []  # type: List[QueuedRequests]
-
-        # Load failed events from queue into failed_requests
-        open(self.queue_file, "a").close()  # Create file if doesn't exist
-        with open(self.queue_file, "r") as queue_fp:
-            for request in queue_fp:
-                logger.debug(request)
-                try:
-                    failed_requests.append(QueuedRequest(*json.loads(request)))
-                except json.decoder.JSONDecodeError as e:
-                    logger.error(e, exc_info=True)
-                    logger.error("Request that failed: {}".format(request))
-                    logger.warning("Skipping request that failed to load")
-
-        # Insert failed_requests into dispatching queue
-        # FIXME: We really shouldn't be clearing the file here until the events have been sent to server.
-        open(self.queue_file, "w").close()  # Clear file
-        if len(failed_requests) > 0:
-            for request in failed_requests:
-                self._queue.put(request)
-            logger.info("Loaded {} failed requests from queuefile".format(len(failed_requests)))
-
-    def _save_queue(self):
-        # When lost connection, save queue to file for later sending
-        with open(self.queue_file, "w") as queue_fp:
-            while not self._queue.empty():
-                request = self._queue.get()
-
-                if request is not None:
-                    queue_fp.write(json.dumps(request) + "\n")
-
     def _try_connect(self) -> bool:
         try:  # Try to connect
             self._create_buckets()
             self.connected = True
-            logger.info("Connection to aw-server established")
+            logger.info("Connection to aw-server established by {}".format(self.client.client_name))
```

Comment:
Why was this change useful? Seems unnecessary to have it print the client name here?

Reply:
In the aw-qt log you couldn't tell which client had connected to the server.

Reply:
Hm, that sucks. It's a shame that python doesn't print the loggers name in the logmessage by default, it it often very useful.
GStreamer does logging very well with the loggers name, then you can filter them with simple env variables such as GST_DEBUG=mylogger:debug,mylogger2:warn etc.. Very neat.

Reply:
I'm pretty sure we could set something similar up in `aw-core`'s `setup_logging`.